client/lib/performance-tracking/components/performance-tracker-stop.jsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { isSingleUserSite, isJetpackSite } from 'state/sites/selectors';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import isSiteStore from 'state/selectors/is-site-store';
+import { stopNavigation } from '../api';
+
+const PerformanceTrackerStop = ( { single, at, jetpack, store } ) => {
```

Comment:
Nit: after merging this it would be nice to add a README for the new lib explaining how to correctly use `PerformanceTrackerStop` and how to decide where to place it in the component tree.